src/main/java/com/atlassian/db/replica/api/ThrottledCache.java

```diff
@@ -29,12 +45,26 @@
     }
 
     private void maybeRefresh(Supplier<T> supplier) {
-        if(lock.tryLock()) {
-            try {
-                value = supplier.get();
-            } finally {
-                lock.unlock();
+        final boolean hasLock = lock.compareAndSet(null, Instant.now().plus(timeout));
+        if (hasLock) {
+            loadValue(supplier);
+        } else {
+            final Instant instant = lock.get();
+            final boolean didLockExpire = instant != null && instant.isBefore(Instant.now());
+            if (didLockExpire) {
+                final boolean didLockTakeOver = lock.compareAndSet(instant, Instant.now().plus(timeout));
+                if (didLockTakeOver) {
```

Comment:
no coverage in test for else  I mean: `didLockTakeOver = false`

Reply:
This case is hard to reproduce (we lose a race condition and another thread started to populate the cache). It would test if atomic reference keeps the contract. I'm not sure if there's enough value in that case to add a test case. 🤔

Reply:
> It would test if atomic reference keeps the contract